### DISCUSSION TITLE
Assignment 1 Clarification

### DISCUSSION THREAD

**Discussion by anushehchaudry** [CS342/.github#8] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

Hi! I was wondering if I could get more clarification about this part in the assignment description?

"The code must conform to the REUSE specification as verified by the REUSE Compliance Check GitHub Action.                                             
The code must adhere to the bundled SwiftLint code formatting specification as verified by the SwiftLint GitHub Action. You are not allowed to modify the SwiftLint configuration.
Disabling a SwiftLint warning should not be used."

More specifically I was wondering what conforming to the REUSE specification means and what was meant when stated that we aren't allowed to modify the Swift Lint configuration. Thank you so much!

### Reproduction

N/A

### Expected behavior

N/A

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment, marked as answer]
Hi @anushehchaudry, happy to help explain this:

[REUSE](https://reuse.software/spec/) defines a standardized method for declaring copyright and licensing in software projects. REUSE makes it clear and easy for people who use our code to understand how each file in the project is licensed. If you look at one of the code files in the `CS342-2023-Application` package, you'll see a REUSE header at the top, which looks like this:

```
//
// This source file is part of the CS342 2023 Application project
//
// SPDX-FileCopyrightText: 2023 Stanford University
//
// SPDX-License-Identifier: MIT
//
```

To comply, you would simply need to include the REUSE header on each new file that is added to the project. There is a Github actions script that runs when you create a pull request to the repository that will check to make sure that each file complies with REUSE and will give you an error if the header is missing.

[SwiftLint](https://github.com/realm/SwiftLint) is a tool that we use to improve the quality of our code by checking it against certain style guidelines. The specific styles/conventions that we have chosen for this project are configured in the [.swiftlint.yml](https://github.com/CS342/CS342-2023-Application/blob/main/.swiftlint.yml) file at the root of the project. You can install and run [SwiftLint](https://github.com/realm/SwiftLint) locally to check to make sure your code follows the guidelines. There is a script that will run Swiftlint when you submit your assignment to check it automatically. Since everyone's code should meet the same guidelines, we don't allow modifying the swiftlint configuration file.

Hope that helps!
Vishnu

**Reply by anushehchaudry** [reply]
Thank you so much for the clarification!
